engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlanConfiguration.java

```diff
@@ -937,13 +1640,21 @@ public LogicalPlanConfiguration addFromProperties(Properties props, Configuratio
       String propertyValue = props.getProperty(propertyName);
       this.properties.setProperty(propertyName, propertyValue);
       if (propertyName.startsWith(StreamingApplication.DT_PREFIX)) {
-        String[] keyComps = propertyName.split("\\.");
+        String[] keyComps = propertyName.split(KEY_SEPARATOR_SPLIT_REGEX);
         parseStramPropertyTokens(keyComps, 1, propertyName, propertyValue, stramConf);
       }
     }
     return this;
   }
 
+  /**
+   * This method is used to parse an Apex property name.
+   * @param keys The keys into which an Apex property is split into.
+   * @param index The current index that the parser is on for processing the property name.
+   * @param propertyName The original unsplit Apex property name.
+   * @param propertyValue The value corresponding to the Apex property.
+   * @param conf
+   */
   private void parseStramPropertyTokens(String[] keys, int index, String propertyName, String propertyValue, Conf conf) {
```

Comment:
This method is now huge, can it be refactored to be easier to understand // be more modular?